### PR TITLE
fix: Allow users to remove linea from networks list

### PR DESCRIPTION
**Pull request by gambinish**
## **Description**

Users should be allowed to remove Linea from networks list. When Linea is removed from default network list, it should appear within the Popular Networks section (be suggested to them). So, we need to also add it to `FEATURED_RPCS`

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27512?quickstart=1)

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/MMASSETS-387

## **Manual testing steps**

1. Go to networks list, ensure that you can remove linea, even though it is a default network.

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by andreahaku — DISMISSED** [review]
LGTM

**Comment by bergeron** [comment]
Let's make sure if linea is deleted, it appears as a popular network so people have the ability to re-add it

**Comment by gambinish** [comment]
> Let's make sure if linea is deleted, it appears as a popular network so people have the ability to re-add it. The list of popular networks should be defined by `FEATURED_RPCS` so you can start with trying to add it there and see what breaks.


https://github.com/user-attachments/assets/742db780-91e0-46ab-9f8d-78a21f22c766

As far as I can tell, this is working as expected since Linea is already added as an entry of `FEATURED_RPCS`

**Comment by gambinish** [comment]
My bad. I had the changes to `FEATURED_RPCS` on my local 😑 

Should be updated now.

File: shared/constants/network.ts [hunk 1]
```diff
@@ -934,7 +934,7 @@ export const FEATURED_RPCS: AddNetworkFields[] = [
     nativeCurrency: CURRENCY_SYMBOLS.ETH,
     rpcEndpoints: [
       {
-        url: LINEA_MAINNET_RPC_URL,
+        url: `https://linea-mainnet.infura.io/v3/${infuraProjectId}`,
         type: RpcEndpointType.Custom,
```
Comment (sahar-fehri) [inline shared/constants/network.ts:938]: Sorry dnt see why are we not using `LINEA_MAINNET_RPC_URL` here?

Comment (gambinish) [reply shared/constants/network.ts:938]: This was actually Brian's feedback from Slack. The idea was to keep the configuration as close to the other ones here in this configuration. The other linea related rpcs are string literals like this.

**Review by bergeron — APPROVED** [review]

**Review by sahar-fehri — APPROVED** [review]

**Review by vinnyhoward — APPROVED** [review]

**Pull request merged**